src/Raven.Client/Documents/Session/Loaders/IIncludeBuilder.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Raven.Client.Extensions;
+using Raven.Client.Documents.Queries.Facets;
+
+namespace Raven.Client.Documents.Session.Loaders
+{
+    public interface IIncludeOperations<T>
+    {
+        IIncludeOperations<T> Counter(string name);
+
+        IIncludeOperations<T> Counters(string[] names);
+
+        IIncludeOperations<T> AllCounters();
+
+        IIncludeOperations<T> Documents(string path);
+
+        IIncludeOperations<T> Documents(Expression<Func<T, string>> path);
+
+        IIncludeOperations<T> Documents(Expression<Func<T, IEnumerable<string>>> path);
+
+    }
+
+    public interface IIncludeBuilder<T>
+    {
+        IIncludeOperations<T> Counter(string name);
+
+        IIncludeOperations<T> Counters(string[] names);
+
+        IIncludeOperations<T> AllCounters();
+
+        IIncludeOperations<T> Documents(string path);
+
+        IIncludeOperations<T> Documents(Expression<Func<T, string>> path);
+
+        IIncludeOperations<T> Documents(Expression<Func<T, IEnumerable<string>>> path);
```

Comment:
We need to cover all cases from https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/IDocumentSession.Include.cs

Reply:
This is a different thing, check this test:
https://github.com/ravendb/ravendb/blob/v4.0/test/SlowTests/MailingList/TestRavenIncludes.cs
and this issue:
http://issues.hibernatingrhinos.com/issue/RavenDB-10968

Reply:
done